src/tiled/command.cpp

```diff
@@ -140,17 +144,20 @@ Command Command::fromQVariant(const QVariant &variant)
         command.workingDirectory = hash[workingDirectoryPref].toString();
     if (hash.contains(shortcutPref))
         command.shortcut = hash[shortcutPref].value<QKeySequence>();
+    if (hash.contains(showOutputPref))
+        command.showOutput = hash[showOutputPref].toBool();
     if (hash.contains(saveBeforeExecutePref))
         command.saveBeforeExecute = hash[saveBeforeExecutePref].toBool();
 
     return command;
 }
 
-CommandProcess::CommandProcess(const Command &command, bool inTerminal)
+CommandProcess::CommandProcess(const Command &command, bool inTerminal, bool showOutput)
     : QProcess(DocumentManager::instance())
     , mName(command.name)
     , mFinalCommand(command.finalCommand())
     , mFinalWorkingDirectory(command.finalWorkingDirectory())
+    , mLogger(CommandManager::instance()->logger())
```

Comment:
If the logger is available from the global `CommandManager` instance, I don't see a reason to also have it as member of `CommandProcess`.